### PR TITLE
CORTX-30595: add addb logs to rgw support bundle

### PR DESCRIPTION
**Pull request by rohit-k-dwivedi**
Signed-off-by: Rohit Dwivedi <rohit.k.dwivedi@seagate.com>

# Problem Statement
- Add addb logs to rgw support bundle
JIRA: https://jts.seagate.com/browse/CORTX-30595
# Design
-  added rgw logs to support bundle
# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by rohit-k-dwivedi** [comment]
demo: dummy addb files for 2 instances of motor
```
[root@cortx-server-headless-svc-ssc-vm-g2-rhev4-3058 rgw]# ls -l /etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/*
-rw-r--r-- 1 root root   5425 May 10 19:37 /etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/radosgw-admin.log
-rw-r--r-- 1 root root 285925 May 10 19:39 /etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/rgw-1.log
-rw-r--r-- 1 root root   8923 May 10 19:37 /etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/rgw_setup.log
-rw-r--r-- 1 root root   1331 May 11 08:11 /etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/rgw_support_bundle.log

'/etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/addb_files-0x7200000000000001:0x30':
total 0
-rw-r--r-- 1 root root 0 May 11 07:41 1_addb.file
-rw-r--r-- 1 root root 0 May 11 07:41 2_addb.file

'/etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/addb_files-0x7300000000000001:0x30':
total 0
-rw-r--r-- 1 root root 0 May 11 07:42 11_abbd.file
-rw-r--r-- 1 root root 0 May 11 07:42 12_abbd.file

/etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/motr_trace_files:
total 32896
-rwx------ 1 root root 16842752 May 10 19:39 m0trace.1
-rwx------ 1 root root 16842752 May 10 19:37 m0trace.346
[root@cortx-server-headless-svc-ssc-vm-g2-rhev4-3058 rgw]#

```

support bundle
```
[root@cortx-server-headless-svc-ssc-vm-g2-rhev4-3058 rgw]# tar -tvf rgw_444774.tar.gz
tar: Removing leading `/' from member names
drwxr-xr-x root/root         0 2022-05-11 08:11 /
-rw-r--r-- root/root      1754 2022-05-11 08:11 cortx_rgw.conf
-rw-r--r-- root/root      5425 2022-05-11 08:11 radosgw-admin.log
-rw-r--r-- root/root      8923 2022-05-11 08:11 rgw_setup.log
drwxr-xr-x root/root         0 2022-05-11 07:42 addb_files-0x7300000000000001:0x30/
-rw-r--r-- root/root         0 2022-05-11 07:42 addb_files-0x7300000000000001:0x30/12_abbd.file
-rw-r--r-- root/root         0 2022-05-11 07:42 addb_files-0x7300000000000001:0x30/11_abbd.file
drwxr-xr-x root/root         0 2022-05-11 07:41 addb_files-0x7200000000000001:0x30/
-rw-r--r-- root/root         0 2022-05-11 07:41 addb_files-0x7200000000000001:0x30/2_addb.file
-rw-r--r-- root/root         0 2022-05-11 07:41 addb_files-0x7200000000000001:0x30/1_addb.file
-rw-r--r-- root/root     18436 2022-05-11 08:11 cortx-rpms
-rw-r--r-- root/root    285925 2022-05-11 08:11 rgw-1.log
-rw-r--r-- root/root      1331 2022-05-11 08:11 rgw_support_bundle.log
[root@cortx-server-headless-svc-ssc-vm-g2-rhev4-3058 rgw]# ls -l /etc/cortx/rgw

```

**Review by sachitanands — APPROVED** [review]
looks good to me

**Review by shriya-deshmukh — APPROVED** [review]

File: src/rgw/support/rgw_support_bundle [hunk 1]
```diff
@@ -81,6 +82,13 @@ class RGWSupportBundle:
         else:
             Log.error("RGW log file does not exists hence skipping log file collection.")
 
+        # copy addb logs
+        addb_log_path = os.path.join(log_dir, 'addb_files-*')
```
Comment (cdeshmukh) [inline src/rgw/support/rgw_support_bundle:86]: whats the log_dir value here. and why cant we copy dir where addb files are be present ?

Add some examples plz. 
Also add plugin is not enabled yet, so with this PR merge even, make sure no issue with support bundle generation is seen.

Comment (rohit-k-dwivedi) [reply src/rgw/support/rgw_support_bundle:86]: log_dir is mentioned in line 74 above 
` log_dir = os.path.join(log_base, f'rgw/{machine_id}') `
 even if  the plugin is unavailable , the code wont break anything as 
 `addb_dirs = glob.glob(addb_log_path)` will return an empty list [] 
        and no opertaion will take place as nothing to iterate in for loop

Comment (cdeshmukh) [reply src/rgw/support/rgw_support_bundle:86]: > and why cant we copy dir where addb files are be present ?

what about this. 
also addb files will be scattered in '/etc/cortx/log/rgw/{machine_id}' dir ?

Comment (sachitanands) [reply src/rgw/support/rgw_support_bundle:86]: > > and why cant we copy dir where addb files are be present ?
> 
> what about this. also addb files will be scattered in '/etc/cortx/log/rgw/{machine_id}' dir ?

in last sprint we created addb directory as a part of our mini-provisioner deployment. (https://github.com/Seagate/cortx-rgw-integration/pull/77)
hence addb directory we should use is " /etc/cortx/log/rgw/{machine-id}/addb_files-{motr fid value}/" and inside this we are expecting addb files to be generated.

Comment (rohit-k-dwivedi) [reply src/rgw/support/rgw_support_bundle:86]: @cdeshmukh  depicted the same in the demo above
create 2 files each in for testing as mentioned by @sachitanands 

'/etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/addb_files-0x7200000000000001:0x30':
total 0
-rw-r--r-- 1 root root 0 May 11 07:41 1_addb.file
-rw-r--r-- 1 root root 0 May 11 07:41 2_addb.file

'/etc/cortx/log/rgw/e6d3c4dcb0574a27aa218003c7f14a6b/addb_files-0x7300000000000001:0x30':
total 0
-rw-r--r-- 1 root root 0 May 11 07:42 11_abbd.file
-rw-r--r-- 1 root root 0 May 11 07:42 12_abbd.file

Comment (cdeshmukh) [reply src/rgw/support/rgw_support_bundle:86]: "addb_files-{motr fid value}" is a directory

**Comment by cdeshmukh** [comment]
rebase PR

**Comment by rohit-k-dwivedi** [comment]
@cdeshmukh rebased

**Pull request merged**